cpp/CMakeLists.txt

```diff
@@ -777,7 +779,7 @@ add_dependencies(cudf jitify_preprocess_run)
 target_link_libraries(
   cudf
   PUBLIC ${ARROW_LIBRARIES} CCCL::CCCL rmm::rmm
-  PRIVATE cuco::cuco ZLIB::ZLIB nvcomp::nvcomp kvikio::kvikio
+  PRIVATE nvtx3-cpp cuco::cuco ZLIB::ZLIB nvcomp::nvcomp kvikio::kvikio
```

Comment:
Will need to also have the cudf tests use this target, since some tests are including internal cudf headers.

Reply:
@robertmaynard I had to do this for benchmarks, too. On second thought, should cudf be exporting this dependency somehow? Or do we expect every library using `cudf` to add `nvtx3-cpp` to their `target_link_libraries`?

Reply:
The reason that tests and benchmarks need to add the `nvtx3-cpp` target is that they include non 'public` headers from `cudf/detail`. 

Actual consumers of libcudf will not do so, and therefore have no need to have nvtx3 installed when building against libcudf.

Reply:
Ah, that makes sense. Thanks, I hadn’t considered that.

Reply:
For posterity, the JNI code was also consuming cudf detail headers and needed to link to NVTX.